Completion.py

```diff
@@ -111,14 +114,18 @@ def read_settings():
         settings["replace_file_path"] = (replace["from"], replace["to"])
     return settings
 
-
-def is_cpp(view):
-    '''Determine if the given view location is c++ code'''
-    try:
-        return view.match_selector(view.sel()[0].begin(), 'source.c++')
-    except:
-        return False
-
+def lang(view):
+    global USER_LANGUAGES
+    if USER_LANGUAGES is None:
+        USER_LANGUAGES = read_settings()['languages']
```

Comment:
I suggest extracting load and verification of languages to separate function. With this approach user will get error messages only on settings reload, not on every plugin activation.